pandas/io/parsers.py

```diff
@@ -769,18 +787,27 @@ def __init__(self, f, engine=None, **kwds):
                 except AttributeError:
                     raise ValueError("Invalid dialect '{dialect}' provided"
                                      .format(dialect=kwds['dialect']))
-                provided = kwds.get(param, _parser_defaults[param])
+                parser_default = _parser_defaults[param]
+                provided = kwds.get(param, parser_default)
 
-                # Messages for conflicting values between the dialect instance
-                # and the actual parameters provided.
+                # Messages for conflicting values between the dialect
+                # instance and the actual parameters provided.
                 conflict_msgs = []
 
-                if dialect_val != provided:
-                    conflict_msgs.append((
-                        "Conflicting values for '{param}': '{val}' was "
-                        "provided, but the dialect specifies '{diaval}'. "
-                        "Using the dialect-specified value.".format(
-                            param=param, val=provided, diaval=dialect_val)))
+                # Don't warn if the default parameter was passed in,
+                # even if it conflicts with the dialect (gh-23761).
+                if provided != parser_default and provided != dialect_val:
+                    msg = ("Conflicting values for '{param}': '{val}' was "
+                           "provided, but the dialect specifies '{diaval}'. "
+                           "Using the dialect-specified value.".format(
+                               param=param, val=provided, diaval=dialect_val))
+
+                    # Annoying corner case for not warning about
+                    # conflicts between dialect and delimiter parameter.
```

Comment:
is there anyway to do this check here, but avoid having all of 'sep_override' stuff. I know this is a PITA.

Reply:
PITA for sure.  Unfortunately, I couldn't come up with anything simpler.  The aliasing between `sep` and `delimiter` makes this difficult to simplify compared to other parameters.  Some sort of special casing has to come through.

I did try modifying `_parser_defaults`, but multiple `read_*` functions rely on the defaults being as they are (i.e. they are test failures).  Thus, I found it more prudent to localize the change to just `read_csv` and not touch other read methods for just one parameter.

As the IO interface cleans up more, we probably will find a better way to do this, but as it stands, I'm willing to stick by this for now.